### PR TITLE
[config] fix: constrain busy threshold range

### PR DESCRIPTION
**Pull request by Wangmerlyn**
## Summary
- constrain `busy_threshold` to `-1` or inclusive `0..100` in the central validator
- validate CLI start/blocking threshold values before service startup or controller construction
- document the percentage range in AGENTS.md, README, CLI/MCP/Python guides, architecture notes, and the branch plan

## Verification
- `PYTHONPATH=$PWD/src pytest tests/utilities/test_session_config.py tests/test_cli_thresholds.py tests/test_cli_service_commands.py tests/mcp/test_server.py tests/cuda_controller/test_throttle.py tests/global_controller/test_contract.py -q` -> 70 passed, 1 skipped
- `PYTHONPATH=$PWD/src pytest tests -q` -> 107 passed, 12 skipped
- `PYTHONPATH=$PWD/src mkdocs build` -> built successfully with existing Material/MkDocs and unnav'd docs notices
- `pre-commit run --all-files` -> passed
- `git diff --check` -> passed

## Local worker notes
- Lovelace performed the TDD source/test slice and confirmed `101` failed before the central validator fix, then passed after it.
- I added CLI preflight validation so invalid thresholds do not auto-start the service before rejection.

## Summary by CodeRabbit

* **New Features**
  * Tightened `busy_threshold` handling across the app to accept only `-1` or values from `0` to `100`.
  * Clarified how backoff behaves when telemetry is unavailable, including when the controller sleeps versus runs continuously.

* **Bug Fixes**
  * Invalid threshold values now fail early in the CLI and API flows with clearer error messages.
  * Added coverage for boundary cases like `100` being allowed and `101` being rejected.

### CODE REVIEW

File: tests/test_cli_thresholds.py [hunk 1]
```diff
@@ -27,3 +27,15 @@ def test_apply_legacy_threshold_memory_string():
...
+
+def test_validate_cli_busy_threshold_rejects_legacy_value_above_percent_range():
+    vram, threshold, mode = cli._apply_legacy_threshold("1GiB", "101", -1)
+
+    assert vram == "1GiB"
+    assert threshold == 101
+    assert mode == "busy"
+    with pytest.raises(
+        Exception, match="busy_threshold must be -1 or an integer between 0 and 100"
+    ):
```
[the remark this answers is not in the document]
Comment (Wangmerlyn) [reply tests/test_cli_thresholds.py:42]: Fixed in 3b24023: tests/test_cli_thresholds.py now imports typer and asserts typer.BadParameter for _validate_cli_busy_threshold(). I also narrowed the adjacent _parse_gpu_ids assertion to the same CLI exception contract. Verified with PYTHONPATH=$PWD/src pytest tests/test_cli_thresholds.py -q, PYTHONPATH=$PWD/src pytest tests -q, pre-commit run --all-files, and git diff --check.
